testsuite/gnat2goto/tests/nat_and_pos/nat_and_pos.adb

```diff
@@ -8,16 +8,14 @@ procedure Nat_And_Pos is
      with Annotate => (ASVAT, In_Type_Function),
      Import;
 begin
-   Var := -1;
-   pragma Assert (In_Type_Natural (Var));
-   pragma Assert (In_Type_Positive (Var));
    Var := 0;
    pragma Assert (In_Type_Natural (Var));
-   pragma Assert (In_Type_Positive (Var));
    Var := 1;
    pragma Assert (In_Type_Natural (Var));
    pragma Assert (In_Type_Positive (Var));
    Var := Integer'Last;
    pragma Assert (In_Type_Natural (Var));
    pragma Assert (In_Type_Positive (Var));
+   Var := -1;
+   pragma Assert (In_Type_Natural (Var));  -- This will cause a range check failure
```

Comment:
It should but in the results it doesn't because the failure on line 12 blocks the rest of the execution.  My suggested fix would be to leave the assertions in the original order but

1. Create a local Boolean variable for each one and leave it uninitialised.
2. For each assertion, make it:
```
if Boolean_n then
  pragma Assert (...);
end if;
```

Reply:
assertions have been placed in original order and if statements added as described.